sdk/java/dagger-java-annotation-processor/src/test/resources/io/dagger/gen/entrypoint/entrypoint.java

```diff
@@ -121,7 +121,9 @@ private ModuleID register() throws ExecutionException, DaggerQueryException,
                     dag.function("nullableDefault",
                         dag.typeDef().withKind(TypeDefKind.STRING_KIND))
                         .withDescription("Set a default value in case the user doesn't provide a value and allow for null value.")
-                        .withArg("stringArg", dag.typeDef().withKind(TypeDefKind.STRING_KIND).withOptional(true), new Function.WithArgArguments().withDefaultValue(JSON.from("\"Foo\"")))));
+                        .withArg("stringArg", dag.typeDef().withKind(TypeDefKind.STRING_KIND).withOptional(true), new Function.WithArgArguments().withDefaultValue(JSON.from("\"Foo\""))))
```

Comment:
I didn't notice this before, but isn't this coming from `containerEcho(@Default("Hello Dagger") String stringArg)`? So there shouldn't be a `withOptional(true)`.

Reply:
This one is coming from `public String nullableDefault(@Nullable @Default("Foo") String stringArg)`

The one you mentioned is for instance `public String nonNullableDefault(@Default("default value") String stringArg)` that generates `.withArg("stringArg", dag.typeDef().withKind(TypeDefKind.STRING_KIND), new Function.WithArgArguments().withDefaultValue(JSON.from("\"default value\""))))` with the default value but no optional.